arkane/statmech.py

```diff
@@ -970,7 +969,35 @@ def determine_qm_software(fullpath):
     return software_log
 
 
-def projectRotors(conformer, F, rotors, linear, is_ts):
+def is_linear(coordinates):
+    """
+    Determine whether or not the species is linear from its 3D coordinates
+    First, try to reduce the problem into just two dimensions, use 3D if the problem cannot be reduced
+    `coordinates` is a numpy.array of the species' xyz coordinates
+    """
+    # epsilon is in degrees
+    # (from our experience, linear molecules have precisely 180.0 degrees between all atom triples)
+    epsilon = 0.1
+
+    number_of_atoms = len(coordinates)
+    if number_of_atoms == 1:
+        return False
+    if number_of_atoms == 2:
+        return True
+
+    # A tensor containing all distance vectors in the molecule
+    d = -numpy.array([c[:, numpy.newaxis] - c[numpy.newaxis, :] for c in coordinates.T])
+    for i in range(2, len(coordinates)):
+        if i > 1:
```

Comment:
You don't need this if-statement anymore since `i` now starts at 2.

Reply:
Thanks, removed